### PR TITLE
ruby-devel: new port (latest Ruby), supports PPC

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port to build upstream Ruby.
Restore of PPC support into `ruby31` (needed to build `ruby32` from upstream/master).

Fixes: https://trac.macports.org/ticket/64906 (closed ticket)

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10A190
Xcode 3.2

macOS 10.5.8
Xcode 3.1.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kimuraw** [comment]
ruby-3.2.0 would be released at 2022-12-25. 
@barracuda156 , what are you planning about port:ruby-devel after ruby-3.2 release?
 
(a). become 3.3 (or 4.0)
(b). keep 3.2, conflicts with ruby32
(c). rename to ruby32

I recommend (a).  

If you choose (b) or (c), I prefer the portname to be "ruby32-devel".

**Comment by barracuda156** [comment]
> ruby-3.2.0 would be released at 2022-12-25. @barracuda156 , what are you planning about port:ruby-devel after ruby-3.2 release?
> 
> (a). become 3.3 (or 4.0) (b). keep 3.2, conflicts with ruby32 (c). rename to ruby32
> 
> I recommend (a).
> 
> If you choose (b) or (c), I prefer the portname to be "ruby32-devel".

@kimuraw Yes, I totally agree with you. `ruby-devel` is to track the upstream (reasonably so, we cannot update it every day, of course).

File: lang/ruby31/files/patch-thread_pthread.diff [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
++    __attribute__((weak)) int pthread_threadid_np(pthread_t, uint64_t*);
++#   endif
++    /* Check weakly linked symbol */
++    const bool no_pthread_threadid_np = !&pthread_threadid_np;
++# endif
++    if (no_pthread_threadid_np) {
++        return ULL2NUM(pthread_mach_thread_np(pthread_self()));
++    }
++# ifndef NO_PTHREAD_MACH_THREAD_NP
++    int e = pthread_threadid_np(target_th->nt->thread_id, &tid);
```
Comment (kimuraw) [inline lang/ruby31/files/patch-thread_pthread.diff:38]: compile failed here on intel macOS 12.6.

> In file included from thread.c:352:
> ./thread_pthread.c:1758:44: error: no member named 'nt' in 'struct rb_thread_struct'
>     int e = pthread_threadid_np(target_th->nt->thread_id, &tid);
>                                 ~~~~~~~~~  ^
> 1 error generated.

"target_th->nt->thread_id" would be "target_th->thread_id".
ruby-3.1.2 does not have rb_thread_struct->nt in  vm_core.h.

I confirmed ruby31 passed core tests (run `make TESTS=ruby test-all`) with change to "target_th->thread_id" on intel macOS 12.6.

**Comment by barracuda156** [comment]
@kimuraw Would you like to be added as a co-maintainer to `ruby-devel`? Would be nice.

**Comment by barracuda156** [comment]
@kimuraw Thank you!

**Pull request merged**